manageiq-operator/pkg/helpers/miq-components/network_policies.go

```diff
@@ -0,0 +1,237 @@
+
+		return nil
+	}
+
+	return networkPolicy, f
+}
+
+func NetworkPolicyAllowPostgres(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme, c *client.Client) (*extensionsv1beta1.NetworkPolicy, controllerutil.MutateFn) {
+	networkPolicy := newNetworkPolicy(cr, "-allow-postgres")
+
+	f := func() error {
+		if err := controllerutil.SetControllerReference(cr, networkPolicy, scheme); err != nil {
+			return err
+		}
+		addAppLabel(cr.Spec.AppName, &networkPolicy.ObjectMeta)
+		setIngressPolicyType(networkPolicy)
+
+		networkPolicy.Spec.PodSelector.MatchLabels = map[string]string{"name": "postgresql"}
+
+		pod := orchestratorPod(*c)
+		if pod == nil {
+			return nil
+		}
+
+		ensureIngressRule(networkPolicy)
+		setFirstIngressTCPPort(networkPolicy, 5432)
+		if len(networkPolicy.Spec.Ingress[0].From) != 2 {
+			networkPolicy.Spec.Ingress[0].From = []extensionsv1beta1.NetworkPolicyPeer{
+				extensionsv1beta1.NetworkPolicyPeer{},
+				extensionsv1beta1.NetworkPolicyPeer{},
+			}
+		}
+		orchestratedByLabelKey := cr.Spec.AppName + "-orchestrated-by"
+		orchestratedByLabelValue := pod.Name
+		networkPolicy.Spec.Ingress[0].From[0].PodSelector = &metav1.LabelSelector{}
+		networkPolicy.Spec.Ingress[0].From[0].PodSelector.MatchLabels = map[string]string{"name": "orchestrator"}
+		networkPolicy.Spec.Ingress[0].From[1].PodSelector = &metav1.LabelSelector{}
+		networkPolicy.Spec.Ingress[0].From[1].PodSelector.MatchLabels = map[string]string{orchestratedByLabelKey: orchestratedByLabelValue}
+
+		return nil
+	}
+
+	return networkPolicy, f
+}
+
+func newNetworkPolicy(cr *miqv1alpha1.ManageIQ, name string) *extensionsv1beta1.NetworkPolicy {
+	return &extensionsv1beta1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Spec.AppName + name,
+			Namespace: cr.ObjectMeta.Namespace,
+		},
+	}
+}
+
+func orchestratorPod(c client.Client) *corev1.Pod {
+	podList := &corev1.PodList{}
+	c.List(context.TODO(), podList)
+
+	for _, pod := range podList.Items {
+		if pod.ObjectMeta.Labels["name"] == "orchestrator" {
+			return &pod
+		}
+	}
+
+	return nil
+}
+
+func setIngressPolicyType(networkPolicy *extensionsv1beta1.NetworkPolicy) {
+	if len(networkPolicy.Spec.PolicyTypes) != 1 {
+		networkPolicy.Spec.PolicyTypes = append(networkPolicy.Spec.PolicyTypes, "Ingress")
+	}
+	networkPolicy.Spec.PolicyTypes[0] = "Ingress"
+}
+
+func ensureIngressRule(networkPolicy *extensionsv1beta1.NetworkPolicy) {
+	if len(networkPolicy.Spec.Ingress) != 1 {
+		networkPolicy.Spec.Ingress = []extensionsv1beta1.NetworkPolicyIngressRule{
+			extensionsv1beta1.NetworkPolicyIngressRule{},
+		}
+	}
+}
+
+func setFirstIngressTCPPort(networkPolicy *extensionsv1beta1.NetworkPolicy, port int32) {
+	if len(networkPolicy.Spec.Ingress[0].Ports) != 1 {
+		networkPolicy.Spec.Ingress[0].Ports = []extensionsv1beta1.NetworkPolicyPort{
+			extensionsv1beta1.NetworkPolicyPort{},
+		}
+	}
+	tcp := corev1.ProtocolTCP
+	networkPolicy.Spec.Ingress[0].Ports[0].Protocol = &tcp
```

Comment:
I'm not a Go write, but can this be:

```suggestion
	networkPolicy.Spec.Ingress[0].Ports[0].Protocol = &(corev1.ProtocolTCP)
```

Reply:
Unfortunately not...
```
INFO[0000] Running the operator locally in namespace bdunne-miq. 
# github.com/ManageIQ/manageiq-pods/manageiq-operator/pkg/helpers/miq-components
pkg/helpers/miq-components/network_policies.go:234:52: cannot take the address of "k8s.io/api/core/v1".ProtocolTCP
FATA[0000] Failed to run operator locally: failed to build operator to run locally: failed to exec []string{"go", "build", "-o", "build/_output/bin/manageiq-operator-local", "github.com/ManageIQ/manageiq-pods/manageiq-operator/cmd/manager"}: exit status 2 
```

Reply:
BTW... I'm not an expert either and this also bothers me so if anyone else knows of a way to do it, I'd be happy to change it.

Reply:
what about `networkPolicy.Spec.Ingress[0].Ports[0].Protocol = &corev1.ProtocolTCP{}` ?

Reply:
Nope.
```
INFO[0000] Running the operator locally in namespace bdunne-miq. 
# github.com/ManageIQ/manageiq-pods/manageiq-operator/pkg/helpers/miq-components
pkg/helpers/miq-components/network_policies.go:234:71: "k8s.io/api/core/v1".ProtocolTCP is not a type
FATA[0000] Failed to run operator locally: failed to build operator to run locally: failed to exec []string{"go", "build", "-o", "build/_output/bin/manageiq-operator-local", "github.com/ManageIQ/manageiq-pods/manageiq-operator/cmd/manager"}: exit status 2 
```

Reply:
Ok, It's just a constant https://github.com/kubernetes/api/blob/5dbd1534041f2d0d080a2cf3b124fb095312c948/core/v1/types.go#L928 we can't take an address of a constant thus the assignment to the tcp var needed.